test/parallel/test-repl-preprocess-top-level-await.js

```diff
@@ -29,38 +29,47 @@ const testCases = [
-    '(async () => { void (a = await 1) })()' ],
+    'let a; (async () => { void (a = await 1) })()' ],
   [ 'const a = await 1',
-    '(async () => { void (a = await 1) })()' ],
+    'let a; (async () => { void (a = await 1) })()' ],
   [ 'for (var i = 0; i < 1; ++i) { await i }',
-    '(async () => { for (void (i = 0); i < 1; ++i) { await i } })()' ],
+    'let i; (async () => { for (void (i = 0); i < 1; ++i) { await i } })()' ],
   [ 'for (let i = 0; i < 1; ++i) { await i }',
-    '(async () => { for (let i = 0; i < 1; ++i) { await i } })()' ],
+    'let i; (async () => { for (void (i = 0); i < 1; ++i) { await i } })()' ],
   [ 'var {a} = {a:1}, [b] = [1], {c:{d}} = {c:{d: await 1}}',
-    '(async () => { void ( ({a} = {a:1}), ([b] = [1]), ' +
+    'let a, b, d; (async () => { void ( ({a} = {a:1}), ([b] = [1]), ' +
         '({c:{d}} = {c:{d: await 1}})) })()' ],
+  [ 'let [a, b, c] = await ([1, 2, 3])',
+    'let a, b, c; (async () => { void ([a, b, c] = await ([1, 2, 3])) })()'],
+  [ 'let {a,b,c} = await ({a: 1, b: 2, c: 3})',
+    'let a, b, c; (async () => { void ({a,b,c} = ' +
+        'await ({a: 1, b: 2, c: 3})) })()'],
+  [ 'let {a: [b]} = {a: [await 1]}, [{d}] = [{d: 3}]',
+    'let b, d; (async () => { void ( ({a: [b]} = {a: [await 1]}),' +
+        ' ([{d}] = [{d: 3}])) })()'],
   /* eslint-disable no-template-curly-in-string */
   [ 'console.log(`${(await { a: 1 }).a}`)',
     '(async () => { return (console.log(`${(await { a: 1 }).a}`)) })()' ],
   /* eslint-enable no-template-curly-in-string */
   [ 'await 0; function foo() {}',
-    '(async () => { await 0; foo=function foo() {} })()' ],
+    'let foo; (async () => { await 0; foo=function foo() {} })()' ],
   [ 'await 0; class Foo {}',
-    '(async () => { await 0; Foo=class Foo {} })()' ],
+    'let Foo; (async () => { await 0; Foo=class Foo {} })()' ],
   [ 'if (await true) { function foo() {} }',
-    '(async () => { if (await true) { foo=function foo() {} } })()' ],
+    'let foo; (async () => { if (await true) { foo=function foo() {} } })()' ],
   [ 'if (await true) { class Foo{} }',
-    '(async () => { if (await true) { class Foo{} } })()' ],
+    'let Foo; (async () => { if (await true) { Foo=class Foo{} } })()' ],
   [ 'if (await true) { var a = 1; }',
-    '(async () => { if (await true) { void (a = 1); } })()' ],
+    'let a; (async () => { if (await true) { void (a = 1); } })()' ],
   [ 'if (await true) { let a = 1; }',
-    '(async () => { if (await true) { let a = 1; } })()' ],
+    'let a; (async () => { if (await true) { void (a = 1); } })()' ],
   [ 'var a = await 1; let b = 2; const c = 3;',
-    '(async () => { void (a = await 1); void (b = 2); void (c = 3); })()' ],
+    'let c; let b; let a; (async () => { void (a = await 1); void (b = 2);' +
+        ' void (c = 3); })()' ],
   [ 'let o = await 1, p',
-    '(async () => { void ( (o = await 1), (p=undefined)) })()' ],
+    'let o, p; (async () => { void ( (o = await 1), (p=undefined)) })()' ],
```

Comment:
It should be to add these:

```suggestion
    'let o, p; (async () => { void ( (o = await 1), (p=undefined)) })()' ],
  [ 'await (async () => { let p = await 1; return p; })()',
    '(async () => { await (async () => { let p = await 1; return p; })() })()' ],
  [ '{ let p = await 1; }',
    '(async () => { { let p = await 1; } })()' ],
  [ 'var p = await 1',
    'var p; (async () => { void (p = await 1) })()' ],
  [ 'await (async () => { var p = await 1; return p; })()',
    '(async () => { await (async () => { var p = await 1; return p; })() })()' ],
  [ '{ var p = await 1; }',
    'var p; (async () => { void (p = await 1); })()' ],
```

Reply:
Shouldn't this

```typescript
[ 'await (async () => { let p = await 1; return p; })()',
    '(async () => { await (async () => { let p = await 1; return p; })() })()' ]
```

 be `'(async () => { return (await (async () => { let p = await 1; return p; })()) })()'`? else the `1` wouldn't be outputted.

Reply:
Yes definitely, improvement on the return values would be great if possible.

Reply:
Just to get this clear (since previous commits has it but didn't receive any comment), only `const` should be rewritten as `let`, but `var` should be hoisted as `var`, correct?

Reply:
Yes that's correct.

Reply:
I've made the adjustments and added new tests, all is working as expected.